### ISSUE TITLE
[RE-OPENED] - [SIM800] Looping with ERROR AT+CIPRXGET=2,1,63 #69

### ISSUE THREAD

**Issue by lafbrazil** [vshymanskyy/TinyGSM#184]
I am using v0.3.5 and code froze after +- 12 hours;

......
[06:25:54.181] Modem inittialized
[06:25:54.181] AT+CSQ
[06:30:24.567] 
[06:30:24.590] +CSQ: 15,0
[06:30:24.631] 
[06:30:24.631] OK
[06:30:24.631] Sending data throguh GPRS
[06:30:24.649] AT+CGATT?
[06:30:24.668] 
[06:30:24.668] +CGATT: 0
[06:30:24.711] 
[06:30:24.711] OK
[06:30:24.711] Waiting for network...AT+CREG?
[06:30:24.727] 
[06:30:24.727] +CREG: 0,5
[06:30:24.767] 
[06:30:24.767] OK
[06:30:24.792]  OK
[06:30:24.792] tim.com.brAT+CIPSHUT
[06:30:24.808] 
[06:30:24.808] SHUT OK
[06:30:24.828] AT+CGATT=0
[06:30:24.848] 
[06:30:24.872] OK
[06:30:24.872] AT+SAPBR=3,1,"Contype","GPRS"
[06:30:24.907] 
[06:30:24.952] OK
[06:30:24.968] AT+SAPBR=3,1,"APN","tim.com.br"
[06:30:25.007] 
[06:30:25.064] OK
[06:30:25.064] AT+SAPBR=3,1,"USER","XXX"
[06:30:25.107] 
[06:30:25.144] OK
[06:30:25.144] AT+SAPBR=3,1,"PWD","XXX"
[06:30:25.187] 
[06:30:25.224] OK
[06:30:25.224] AT+CGDCONT=1,"IP","tim.com.br"
[06:30:25.272] 
[06:30:25.336] OK
[06:30:25.336] AT+CGACT=1,1
[06:30:25.358] 
[06:30:31.827] OK
[06:30:31.827] AT+SAPBR=1,1
[06:30:31.850] 
[06:30:33.138] OK
[06:30:33.158] AT+SAPBR=2,1
[06:30:33.178] 
[06:30:33.202] +SAPBR: 1,1,"100.88.72.189"
[06:30:33.257] OK
[06:30:33.282] AT+CGATT=1
[06:30:33.282] 
[06:30:33.318] OK
[06:30:33.318] AT+CIPMUX=1
[06:30:33.335] 
[06:30:33.362] OK
[06:30:33.459] AT+CIPQSEND=1
[06:30:33.478] 
[06:30:33.507] OK
[06:30:33.507] AT+CIPRXGET=1
[06:30:33.527] 
[06:30:33.555] OK
[06:30:33.582] AT+CSTT="tim.com.br","XXX","XXX"
[06:30:33.619] 
[06:30:33.683] OK
[06:30:33.683] AT+CIICR
[06:30:33.708] 
[06:30:33.731] OK
[06:30:33.731] AT+CIFSR;E0
[06:30:33.749] 
[06:30:33.788] 100.88.72.189
[06:30:33.811] 
[06:30:33.811] OK
[06:30:33.811] AT+CDNSCFG="8.8.8.8","8.8.4.4"
[06:30:33.847] 
[06:30:33.891] OK
[06:30:33.908] Connected
[06:30:33.908] ---Incoming Data start ... Connection 1
[06:30:33.971] Connection 4
[06:30:33.971] Connection 5
[06:30:33.988] AT+CIPCLOSE=1
[06:30:34.008] 
[06:30:34.008] ERROR
[06:30:34.008] AT+CIPSSL=1
[06:30:34.028] 
[06:30:34.028] OK
[06:30:34.051] AT+CIPSTART=1,"TCP",IP",443
[06:30:34.087] 
[06:30:34.131] OK
[06:30:34.148] 
[06:30:38.458] 1, CONNECT OK
[06:30:38.479] Connected
[06:30:38.499] AT+CIPSEND=1,3
[06:30:38.522] 
[06:30:38.522] >GET 
[06:30:38.560] DATA ACCEPT:1,3
[06:30:38.602] AT+CIPSEND=1,1
[06:30:38.602] 
[06:30:38.634] >  
[06:30:38.668] DATA ACCEPT:1,1
[06:30:38.689] AT+CIPSEND=1,36
[06:30:38.732] 01B343/1.49/13.00/14.48 
[06:30:38.843] DATA ACCEPT:1,36
[06:30:38.878] AT+CIPSEND=1,9
[06:30:38.899] 
[06:30:38.923] > HTTP/1.1 
[06:30:38.971] DATA ACCEPT:1,9
[06:30:39.008] AT+CIPSEND=1,2
[06:30:39.029] 
[06:30:39.051] >
[06:30:39.051]  
[06:30:39.083] DATA ACCEPT:1,2
[06:30:39.118] 01B343/1.49/13.00/14.48 HTTP/1.1
[06:30:39.163] AT+CIPSEND=1,6
[06:30:39.179] 
[06:30:39.179] >Host:  
[06:30:39.199] DATA ACCEPT:1,6
[06:30:39.243] AT+CIPSEND=1,17
[06:30:39.259] 
[06:30:39.299] >IP
[06:30:39.339] DATA ACCEPT:1,17
[06:30:39.468] AT+CIPSEND=1,1
[06:30:39.489] 
[06:30:39.509] >: 
[06:30:39.532] DATA ACCEPT:1,1
[06:30:39.568] AT+CIPSEND=1,3
[06:30:39.589] 
[06:30:39.612] >443 
[06:30:39.648] DATA ACCEPT:1,3
[06:30:39.692] AT+CIPSEND=1,2
[06:30:39.692] 
[06:30:39.724] >
[06:30:39.724]  
[06:30:39.756] DATA ACCEPT:1,2
[06:30:39.799] AT+CIPSEND=1,10
[06:30:39.818] 
[06:30:39.836] >User-Agent 
[06:30:39.884] DATA ACCEPT:1,10
[06:30:39.932] AT+CIPSEND=1,2
[06:30:39.932] 
[06:30:39.964] >:  
[06:30:40.012] DATA ACCEPT:1,2
[06:30:40.029] AT+CIPSEND=1,13
[06:30:40.049] 
[06:30:40.076] >Arduino/2.2.0 
[06:30:40.140] DATA ACCEPT:1,13
[06:30:40.159] AT+CIPSEND=1,2
[06:30:40.179] 
[06:30:40.204] >
[06:30:40.229]  
[06:30:40.252] DATA ACCEPT:1,2
[06:30:40.269] AT+CIPSEND=1,2
[06:30:40.290] 
[06:30:40.317] >
[06:30:40.339]  
[06:30:40.359] DATA ACCEPT:1,2
[06:30:40.478] AT+CIPRXGET=4,1
[06:30:40.502] 
[06:30:40.522] +CIPRXGET: 4,1,0
[06:30:40.543] OK
[06:30:40.562] AT+CIPSTATUS=1
[06:30:40.583] 
[06:30:40.605] +CIPSTATUS: 1,0,"TCP","52.67.255.181","443","CONNECTED"
[06:30:40.719] OK
[06:30:40.740] AT+CIPRXGET=4,1
[06:30:41.747] 
[06:30:41.774] +CIPRXGET: 1,1
[06:30:41.798] 
[06:30:41.798] +CIPRXGET: 4,1,426
[06:30:41.822] 
[06:30:41.822] OK
[06:30:41.839] AT+CIPRXGET=2,1,511
[06:30:41.859] 
[06:30:41.902] +CIPRXGET: 2,1,426,0
[06:30:41.959] HTTP/1.1 200 OK
[06:30:41.983] X-Powered-By: Express
[06:30:42.037] Access-Control-Allow-Origin: *
[06:30:42.111] Content-Type: application/json; charset=utf-8
[06:30:42.191] Content-Length: 20
[06:30:42.227] ETag: W/"14-cbmD/auWoSatriSIgZCfCUgg0jw"
[06:30:42.329] set-cookie: connect.sid=s%3AA1LQI2TAo6FO3cqn2mnYozyl4BtZOcCx.rzIl9Zw4rgODd75EVwk1s2bOGxIIFSaQuLpH1HyHvDQ; Path=/; Expires=Thu, 26 Jul 2018 09:48:12 GMT; HttpOnly
[06:30:42.671] Date: Thu, 26 Jul 2018 09:33:12 GMT
[06:30:42.733] Connection: keep-alive
[06:30:42.767] 
[06:30:42.788] {"message":"SUCCESS"
[06:30:42.832] +CIPRXGET: 1,1
[06:30:42.848] 
[06:30:42.868] OK
[06:30:42.868] AT+CIPRXGET=4,1
[06:30:42.888] 
[06:30:42.912] +CIPRXGET: 4,1,1
[06:30:42.968] 
[06:30:42.968] OK
[06:30:42.968] 200
[06:30:42.992] ---Closing connections ...
[06:30:43.027] AT+CIPCLOSE=1
[06:30:43.048] 
[06:30:43.072] 1, CLOSE OK
[06:30:43.088] Server disconnected
[06:30:53.128] AT+CIPSHUT
[06:30:53.149] 
[06:30:53.149] SHUT OK
[06:30:53.149] AT+CGATT=0
[06:30:53.166] 
[06:30:53.197] +SAPBR 1: DEACT
[06:30:53.259] 
[06:30:54.199] OK
[06:30:54.199] GPRS disconnected
[06:30:54.223] AT
[06:30:54.223] 
[06:30:54.223] OK
[06:30:54.223] AT&FZ
[06:30:54.339] 
[06:30:54.354] OK
[06:30:54.464] ATE0
[06:30:54.464] ATE0
[06:30:54.512] 
[06:30:54.512] OK
[06:30:54.512] AT+CPIN?
[06:30:54.512] 
[06:30:54.528] +CPIN: READY
[06:30:54.568] OK
[06:30:54.592] Modem inittialized
[06:30:54.592] AT+CSQ
[06:35:24.994] 
[06:35:25.041] +CSQ: 15,0
[06:35:25.041] 
[06:35:25.041] OK
[06:35:25.058] Sending data throguh GPRS
[06:35:25.097] AT+CGATT?
[06:35:25.097] 
[06:35:25.097] +CGATT: 0
[06:35:25.122] 
[06:35:25.122] OK
[06:35:25.122] Waiting for network...AT+CREG?
[06:35:25.158] +CREG: 0,5
[06:35:25.202] 
[06:35:25.202] OK
[06:35:25.202]  OK
[06:35:25.202] tim.com.brAT+CIPSHUT
[06:35:25.237] 
[06:35:25.237] SHUT OK
[06:35:25.258] AT+CGATT=0
[06:35:25.278] 
[06:35:25.299] OK
[06:35:25.299] AT+SAPBR=3,1,"Contype","GPRS"
[06:35:25.337] 
[06:35:25.426] OK
[06:35:25.426] AT+SAPBR=3,1,"APN","tim.com.br"
[06:35:25.487] 
[06:35:25.522] OK
[06:35:25.538] AT+SAPBR=3,1,"USER","XXX"
[06:35:25.577] 
[06:35:25.602] OK
[06:35:25.622] AT+SAPBR=3,1,"PWD","XXX"
[06:35:25.657] 
[06:35:25.682] OK
[06:35:25.698] AT+CGDCONT=1,"IP","tim.com.br"
[06:35:25.737] 
[06:35:25.794] OK
[06:35:25.818] AT+CGACT=1,1
[06:35:25.838] 
[06:35:29.431] OK
[06:35:29.431] AT+SAPBR=1,1
[06:35:29.449] 
[06:35:30.684] OK
[06:35:30.709] AT+SAPBR=2,1
[06:35:30.729] 
[06:35:30.729] +SAPBR: 1,1,"100.85.232.229"
[06:35:30.809] 
[06:35:30.809] OK
[06:35:30.809] AT+CGATT=1
[06:35:30.828] 
[06:35:30.848] OK
[06:35:30.869] AT+CIPMUX=1
[06:35:30.889] 
[06:35:30.889] OK
[06:35:30.908] AT+CIPQSEND=1
[06:35:30.928] 
[06:35:30.953] OK
[06:35:30.953] AT+CIPRXGET=1
[06:35:30.978] 
[06:35:31.001] OK
[06:35:31.018] AT+CSTT="tim.com.br","XXX","XXX"
[06:35:31.078] 
[06:35:31.138] OK
[06:35:31.138] AT+CIICR
[06:35:31.161] 
[06:35:31.161] OK
[06:35:31.179] AT+CIFSR;E0
[06:35:31.198] 
[06:35:31.218] 100.85.232.229
[06:35:31.241] OK
[06:35:31.258] AT+CDNSCFG="8.8.8.8","8.8.4.4"
[06:35:31.297] 
[06:35:31.439] OK

############################################### ATTENTION

[06:35:31.439] Connected
[06:35:31.439] ---Incoming Data start ... Connection 1
[06:35:31.482] Connection already open        >>> *******************  ATTENTION
[06:35:31.517] Connected
[06:35:31.517] AT+CIPSEND=1,3
[06:35:31.562] 
[06:35:31.562] ERROR
[06:35:31.562] AT+CIPSEND=1,1
[06:35:31.562] 
[06:35:31.562] ERROR
[06:35:31.578] AT+CIPSEND=1,36
[06:35:31.598] 
[06:35:31.626] ERROR
[06:35:31.648] AT+CIPSEND=1,9
[06:35:31.669] 
[06:35:31.690] ERROR
[06:35:31.722] AT+CIPSEND=1,2
[06:35:31.722] 
[06:35:31.754] ERROR
[06:35:31.778] 01B343/1.49/13.00/14.48 HTTP/1.1
[06:35:31.819] AT+CIPSEND=1,6
[06:35:31.840] 
[06:35:31.840] ERROR
[06:35:31.858] AT+CIPSEND=1,17
[06:35:31.882] 
[06:35:31.882] ERROR
[06:35:31.898] AT+CIPSEND=1,1
[06:35:31.918] 
[06:35:31.946] ERROR
[06:35:31.969] AT+CIPSEND=1,3
[06:35:31.988] 
[06:35:32.011] ERROR
[06:35:32.028] AT+CIPSEND=1,2
[06:35:32.045] 
[06:35:32.091] ERROR
[06:35:32.091] AT+CIPSEND=1,10
[06:35:32.091] 
[06:35:32.123] ERROR
[06:35:32.138] AT+CIPSEND=1,2
[06:35:32.158] 
[06:35:32.187] ERROR
[06:35:32.208] AT+CIPSEND=1,13
[06:35:32.228] 
[06:35:32.251] ERROR
[06:35:32.269] AT+CIPSEND=1,2
[06:35:32.288] 
[06:35:32.315] ERROR
[06:35:32.427] AT+CIPSEND=1,2
[06:35:32.449] 
[06:35:32.475] ERROR
[06:35:32.498]

So, basically after detected that Connection already open the device don't have socket available and froze the device and is required a manual reset.

Please, looking for potential solution or fix;

Regards,
LAF

**Comment by SRGDamia1** [comment]
I'm guessing this is fixed in the latest version.
